cpp/src/parquet/arrow/reader.cc

```diff
@@ -1315,20 +1315,32 @@ Status FileReader::GetRecordBatchReader(const std::vector<int>& row_group_indice
   return Status::OK();
 }
 
-Status FileReader::Make(::arrow::MemoryPool* pool,
-                        std::unique_ptr<ParquetFileReader> reader,
+Status FileReader::Make(MemoryPool* pool, std::unique_ptr<ParquetFileReader> reader,
                         const ArrowReaderProperties& properties,
                         std::unique_ptr<FileReader>* out) {
   *out = std::make_unique<FileReaderImpl>(pool, std::move(reader), properties);
   return static_cast<FileReaderImpl*>(out->get())->Init();
```

Comment:
Can we use the `Result` version here?

```suggestion
  return Make(pool, std::move(reader), properties)->Value(out);
```